Services/Certificate/classes/Portfolio/class.ilPortfolioCertificateFileService.php

```diff
@@ -1,33 +1,23 @@
-<?php
+<?php declare(strict_types=1);
+
 /* Copyright (c) 1998-2019 ILIAS open source, Extended GPL, see docs/LICENSE */
 
 use ILIAS\Filesystem\Filesystem;
+use ILIAS\Filesystem\Exception\FileAlreadyExistsException;
+use ILIAS\Filesystem\Exception\IOException;
+use ILIAS\Filesystem\Exception\FileNotFoundException;
 
 /**
  * @author  Niels Theen <ntheen@databay.de>
  */
 class ilPortfolioCertificateFileService
 {
-    /**
-     * @var Filesystem
-     */
-    private $filesystem;
-
-    /**
-     * @var ilLogger|null
-     */
-    private $logger;
-
+    private ?Filesystem $filesystem;
+    private ?ilLogger $logger;
```

Comment:
These types are not optional. Only the passed constructor arguments are optional.